frontend/js/src/common/Username.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+import useUserFlairs from "../utils/FlairLoader";
+
+function Username({
+  username,
+  elementType,
+  hideFlair = false,
+  showLink = false,
+  ...otherProps
+}: {
+  username: string;
+  elementType?: keyof JSX.IntrinsicElements;
+  hideFlair?: boolean;
+  showLink?: boolean;
```

Comment:
I think it might make more sense to have the default be a link, and the option to be `hideLink` instead when we really don't want it to be a link (which according to my understanding will be more rare).

In that case, perhaps there is a way to make elementType required, but only when hideLink is true?
We have an example of that in the codebase:

https://github.com/metabrainz/listenbrainz-server/blob/90453e5ef35c5b1c21051d9f33882c8e2a8d5b09/frontend/js/src/explore/art-creator/ArtCreator.tsx#L25C1-L40C1

Reply:
Hmm, yes that's a better approach. Fixed